### DISCUSSION TITLE
Question on COCORAHS precip verification

### DISCUSSION THREAD

**Discussion by PerryShafran-NOAA** [dtcenter/METplus#2080] [Use Cases] [answered]
Asking a question that we didn't quite get to in the Monday METplus User's Discussion with EMC/DTC.  Printing it out verbatim from the notes of that meeting (3/6):

We spoke about the COCORAHS precip verification several weeks ago.  I just wanted to check in for the correct settings that we need for precip verification valid at 12Z, because the notes didn’t quite make clear what the correct settings should be.

I have the following:

VALID_BEG = {ENV[VDATE]}12
VALID_END = {ENV[VDATE]}12
VALID_INCREMENT = 1M
LEAD_SEQ = 24

ASCII2NC_FILE_WINDOW_BEGIN = 0
ASCII2NC_FILE_WINDOW_END = 0

FCST_PCP_COMBINE_COMMAND = -sum 00000000_000000 1 {ENV[VDATE]}_120000 24 {FCST_PCP_COMBINE_OUTPUT_DIR}/{FCST_PCP_COMBINE_OUTPUT_TEMPLATE} -pcpdir {FCST_PCP_COMBINE_INPUT_DIR}/{FCST_PCP_COMBINE_INPUT_TEMPLATE}

OBS_POINT_STAT_WINDOW_BEGIN = -82400
OBS_POINT_STAT_WINDOW_END = 0

Please define the first 00000000_000000, what this represents and the 1 following that.

George: See 8. Re-Formatting of Gridded Fields — MET 11.0.1 documentation
Providing a string of all zeros (00000000_000000) indicates that all files, regardless of initialization time should be processed.

I have set {ENV[VDATE]}_120000 24 to be the valid date of the end of the summing period, and the 24 is the hours to be summed.

Somewhere, I want to be sure I have the past 24 hours of valid analysis data to be able to sum that.  Where specifically do I define that.  I put that in the OBS_POINT_STAT_WINDOW variables, but are there PCP_COMBINE window variables, which I believe should be there?  

Thanks for any assistance you can provide.

Perry

**Comment by CPKalb** [comment, marked as answer]
Hi Perry:

I wasn't part of the discussion Monday.  However, when using the sum command, the required input arguments are init_time, in_accum in hours, valid_time, and out_accum in hours.  For observations (rather than model), the init_time should be specified as 00000000_000000, since there isn't a model initialization.  It looks like you have specified your valid_time as {ENV[VDATE]}_120000, in_accum as 1, and out_accum as 24.

With the configurations listed, MET will take the {ENV{VDATE}}_120000 as the end of the 24 hour period, and look for the 24 one hour precip files leading up to that date, in the directory given after pcpdir.   However, it looks like you are running pcp_combine on forecast data.  In this case, you probably want to specify the model initialization as something other than 00000000_000000, so that MET will know which initialization to use for these these 24 one hour files.  If you are not running on forecast data, then you will want to change your PCP_COMBINE variable names to say OBS_PCP_COMBINE rather than FCST_PCP_COMBINE.

Pcp combine does have a [sum](https://metplus.readthedocs.io/en/latest/generated/met_tool_wrapper/PCPCombine/PCPCombine_sum.html) option, which can make setting up these types of runs easier.  It looks like you are using the USER_DEFINED option. 

Does this help? 

Christina

**Comment by CPKalb** [comment]
Hi Perry:

Were you able to get this working as expected?

Thanks,
Christina

**Reply by PerryShafran-NOAA** [reply]
Hi, Christina,

Unfortunately, I've been so busy with other things that I haven't been able to work on this.  But I should add that I am trying to run pcp_combine on 1-hr gridded analyses (which are like models), but I want to make 24-hr accumulations of 24 1-hr files from 12Z to 12Z.  So it's not like a model in that it's a model forecast 24 hours ago, but it's individual sequential 1-hr precip for each hour.  I need to know exactly how to do that, and I was not sure how to use the sample METplus use case configuration file included in the METplus documentation example.  

Perry

**Reply by CPKalb** [reply]
Hi Perry:

Is there only one gridded analysis for each 1 hour time?  If so, then the way you currently have your file should do what you are wanting.  It will make a 24 hour accumulation out of 1 hour files ending at {ENV[VDATE]}_120000. 

Christina

**Reply by JohnHalleyGotway** [reply]
Perry, just following up on this old discussion. Do any questions remain?

**Reply by PerryShafran-NOAA** [reply]
I haven't been able to work on this for awhile, so maybe.  Hold for now.

Perry

**Reply by jprestop** [reply]
@PerryShafran-NOAA  Hi Perry.  I'm just following up on this old Discussion to see if you've had a chance to work on it.  Do any questions remain?

**Reply by PerryShafran-NOAA** [reply]
Unfortunately my development time on this has been limited, but I'm sure that I'll have questions again - especially as I try to remember what I had asked previously.  Sorry to keep this thread hanging...

Thanks!

Perry

**Reply by jprestop** [reply]
That's ok @PerryShafran-NOAA.  We understand. We just want to check in periodically so the Discussions don't get forgotten about.  Thanks for letting us know. We'll leave this one open.

**Reply by JohnHalleyGotway** [reply]
@PerryShafran-NOAA, it looks like this discussion on this CoCoRaHS use case has shifted over to #2220. Recommend that we mark this discussion as being answered and discuss further details on #2220.
